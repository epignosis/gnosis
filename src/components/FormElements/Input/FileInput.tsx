import React, { FC, InputHTMLAttributes } from "react";
import { Label } from "../../../";
import { fileInputContainer } from "./styles";
import { ExtendableProps } from "types/utils";

export type FileInputProps = ExtendableProps<
  InputHTMLAttributes<HTMLInputElement>,
  {
    id: string;
    name: string;
    label?: string;
    type?: never;
    children?: never;
  }
>;

const FileInput: FC<FileInputProps> = ({ id, label, name, ...rest }) => (
  <div css={fileInputContainer}>
    <label htmlFor={id}>
      <Label id={`${id}-${name}`}>{label}</Label>
      <div className="drop-area">
        <input id={id} type="file" name={name} {...rest} aria-describedby={`${id}-${name}`} />
      </div>
    </label>
  </div>
);

export default FileInput;
