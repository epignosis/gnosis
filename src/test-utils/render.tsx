import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
