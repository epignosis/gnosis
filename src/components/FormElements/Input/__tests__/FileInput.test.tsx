import React from "react";
import userEvent from "@testing-library/user-event";
import faker from "faker";
import FileInput from "../FileInput";
import { screen, render } from "@test-utils/render";

describe("<Input />", () => {
  it("renders correctly", () => {
    const labelTxt = faker.lorem.word();
    const fileName = faker.system.fileName();
    const fileType = faker.system.fileExt();

    const file = new File(["test"], fileName, { type: fileType });

    render(<FileInput id="test-input" name="test-file-input" label={labelTxt} />);

    const label = screen.getByText(labelTxt);
    const input = screen.getByLabelText(labelTxt);

    expect(label).toHaveTextContent(labelTxt);
    expect(input).toHaveAttribute("name", "test-file-input");

    userEvent.upload(input, file);

    expect(file.name).toBe(fileName);
    expect(file.type).toBe(fileType);
  });

  it("matches snapshot", () => {
    const { container } = render(
      <FileInput id="test-input" name="test-file-input" label="test label" />,
    );

    expect(container).toMatchSnapshot();
  });
});
