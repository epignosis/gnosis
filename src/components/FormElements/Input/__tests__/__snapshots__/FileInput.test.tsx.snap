// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> matches snapshot 1`] = `
<div>
  <div
    class="css-n9rd9a-fileInputContainer-fileInputContainer"
  >
    <label
      for="test-input"
    >
      <label
        class="css-xs6o3g-label-label"
        id="test-input-test-file-input"
      >
        test label
      </label>
      <div
        class="drop-area"
      >
        <input
          aria-describedby="test-input-test-file-input"
          id="test-input"
          name="test-file-input"
          type="file"
        />
      </div>
    </label>
  </div>
</div>
`;
